packages/react-utilities/src/compose/getSlots.ts

```diff
@@ -44,7 +34,10 @@ export function getSlots<SlotProps extends SlotPropsRecord = {}>(state: Componen
   type Slots = { [K in keyof SlotProps]: React.ElementType<SlotProps[K]> };
 
   const slots = ({
-    root: getSlot(typedState.components ? typedState.components.root : undefined, typedState.as),
+    root:
+      typedState.components?.root === undefined || typeof typedState.components.root === 'string'
+        ? typedState.as || typedState.components?.root || 'div'
+        : typedState.components.root,
```

Comment:
```suggestion
    // If components.root is a React component we should pass down "as" prop to it
    root:
      typedState.components?.root === undefined || typeof typedState.components.root === 'string'
        ? typedState.as || typedState.components?.root || 'div'
        : typedState.components.root,
```

Can we add a comment? The function was more readable, can you please restore it?